### PR TITLE
[SDK-4218] Flutter Web quickstart

### PR DESCRIPTION
**Pull request by stevehobbsdev**
This PR adds a new quickstart for Flutter Web. It's mostly a copy of the [native quickstart](https://auth0.com/docs/quickstart/native/flutter/interactive) but tuned for the web platform.

image

### CODE REVIEW

File: articles/quickstart/spa/flutter/01-login.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    child: const Text("Log in")
+  )
+}
+```
+
+When a user logs in, they are redirected back to your application. You are then able to access the ID and access tokens for this user by calling `onLoad` during startup and handling the credentials that are given to you:
+
+```dart
+auth0.onLoad().then((final credentials) => setState(() {
+    // handle or store credentials here
```
Comment (Widcket) [inline articles/quickstart/spa/flutter/01-login.md]:
```suggestion
    // Handle or store credentials here
    _credentials = credentials;
```
Let's make it complete

File: articles/quickstart/spa/flutter/01-login.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+:::checkpoint-default
+Add a button to your app that calls `loginWithRedirect()` and logs the user into your app. Verify that you are redirected to Auth0 for authentication and then back to your application.
+
+Verify that you can access `credentials` as a result of calling `onLoad` and that you're able to access the ID and access tokens.
+:::
+
+:::checkpoint-failure
+If your application did not launch successfully:
+
+- Ensure you set the Allowed Callback URLs are correct
```
Comment (Widcket) [inline articles/quickstart/spa/flutter/01-login.md]:
```suggestion
- Ensure the Allowed Callback URLs are set properly
```

File: articles/quickstart/spa/flutter/01-login.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+```
+
+::::checkpoint
+:::checkpoint-default
+Log in and inspect the `user` property on the result. Verify the current user's profile information, such as `email` or `name`.
+:::
+:::checkpoint-failure
+If your application did not return user profile information:
+
+- Verify the access token is valid
```
Comment (Widcket) [inline articles/quickstart/spa/flutter/01-login.md]:
```suggestion
- Verify the ID token is valid
```

File: articles/quickstart/spa/flutter/files/main.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        else
+          Column(
+            children: [
+              ProfileView(user: _credentials!.user),
+              ElevatedButton(
+                  onPressed: () async {
+                    setState(() {
+                      _credentials = null;
+                    });
+
```
Comment (Widcket) [inline articles/quickstart/spa/flutter/files/main.md]:
```suggestion
```
I think this is not actually necessary, as after the redirect onLoad() will kick in. Will remove it from the sample app as well (will leave it only for mobile).

File: articles/quickstart/spa/flutter/01-login.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+See this example of an `ElevatedButton` widget that logs the user out of the app. Note that `_credentials` is set to `null`, indicating that the user is no longer signed in to your app:
+
+```dart
+ElevatedButton(
+  onPressed: () async {
+    setState(() {
+      _credentials = null;
+    });
+
```
Comment (Widcket) [inline articles/quickstart/spa/flutter/01-login.md]:
```suggestion
```
I think this is not actually necessary, as after the redirect onLoad() will kick in. Will remove it from the sample app as well (will leave it only for mobile).

File: articles/quickstart/spa/flutter/index.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+hidden_articles:
+  - interactive
+show_steps: true
+github:
+  org: auth0-samples
+  repo: auth0-flutter-samples
+  branch: main
+requirements:
+  - Flutter 3+
+  - Xcode 13+ (for iOS)
```
Comment (Widcket) [inline articles/quickstart/spa/flutter/index.yml]:
```suggestion
  - Xcode 14+ (for iOS)
```
It's 14+ nowadays.

File: articles/quickstart/spa/flutter/interactive.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+:::note
+You will normally need to specify the `redirectUrl` parameter to `loginWithRedirect`. Omitting this will cause Auth0 to use the [default login route](https://auth0.com/docs/authenticate/login/auth0-universal-login/configure-default-login-routes), which is not configured by default.
+:::
+
+When a user logs in, they are redirected back to your application. You are then able to access the ID and access tokens for this user by calling `onLoad` during startup and handling the credentials that are given to you:
+
+```dart
+auth0.onLoad().then((final credentials) => setState(() {
+    // handle or store credentials here
```
Comment (Widcket) [inline articles/quickstart/spa/flutter/interactive.md]:
```suggestion
    // Handle or store credentials here
    _credentials = credentials;
```

File: articles/quickstart/spa/flutter/interactive.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+:::checkpoint-default
+Add a button to your app that calls `loginWithRedirect()` and logs the user into your app. Verify that you are redirected to Auth0 for authentication and then back to your application.
+
+Verify that you can access `credentials` as a result of calling `onLoad` and that you're able to access the ID and access tokens.
+:::
+
+:::checkpoint-failure
+If your application did not launch successfully:
+
+- Ensure you set the Allowed Callback URLs are correct
```
Comment (Widcket) [inline articles/quickstart/spa/flutter/interactive.md]:
```suggestion
- Ensure the Allowed Callback URLs are set properly
```

File: articles/quickstart/spa/flutter/interactive.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+The user profile automatically retrieves user profile properties for you when the page loads, and can be accessed and stored by calling `onLoad` during application startup. The returned object from `onLoad` contains a `user` property with all the user profile properties. This is internally populated by decoding the ID token.
+
+::::checkpoint
+:::checkpoint-default
+Log in and inspect the `user` property on the result. Verify the current user's profile information, such as `email` or `name`.
+:::
+:::checkpoint-failure
+If your application did not return user profile information:
+
+- Verify the access token is valid
```
Comment (Widcket) [inline articles/quickstart/spa/flutter/interactive.md]:
```suggestion
- Verify the ID token is valid
```

File: articles/quickstart/spa/flutter/01-login.md (hunk 5 above)
Comment (stevehobbsdev) [reply articles/quickstart/spa/flutter/01-login.md]: I agree

**Review by Widcket — DISMISSED** [review]

File: articles/quickstart/spa/flutter/01-login.md [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+Still having issues? Check out our [documentation](https://auth0.com/docs) or visit our [community page](https://community.auth0.com) to get more help.
+:::
+::::
+
+## Add logout to your app
+
+To log users out, redirect them to the Auth0 logout endpoint to clear their login session by calling the Auth0 Flutter SDK `logout()`. [Read more about logging out of Auth0](https://auth0.com/docs/authenticate/login/logout).
+
+See this example of an `ElevatedButton` widget that logs the user out of the app. Note that `_credentials` is set to `null`, indicating that the user is no longer signed in to your app:
```
Comment (Widcket) [inline articles/quickstart/spa/flutter/01-login.md]:
```suggestion
See this example of an `ElevatedButton` widget that logs the user out of the app:
```

**Review by Widcket — DISMISSED** [review]

**Review by Widcket — APPROVED** [review]

**Pull request merged**